File: app/src/main/java/ds/features/binding/WeatherViewModel.java
package ds.features.binding;

import android.databinding.BindingAdapter;
import android.view.View;
import android.widget.ImageView;
import com.squareup.picasso.Picasso;
import ds.features.Constants;
import ds.features.db.DB;
import ds.features.db.realm.Weather;

public class WeatherViewModel {

	public Weather data;


	public WeatherViewModel(final long id) {
		this.data = DB.Companion.getInstance().getWeatherModel(id);
	}

	public WeatherViewModel(Weather data) {
		this.data = data;
	}


	public int humidityVisibility() {
		return data.getHumidity() == null ? View.GONE : View.VISIBLE;
	}


	@BindingAdapter("app:iconUrl")
	public static void iconUrl(ImageView img, String url) {
		Picasso.with(img.getContext()).load(String.format(Constants.IMAGES_BASE_URL, url)).into(img);
	}


	@BindingAdapter("app:windDirection")
	public static void windDirection(ImageView img, float deg) {
		img.setRotation(-90);
		img.animate().rotationBy(deg);
	}


	@BindingAdapter("app:windDirectionNow")
	public static void windDirectionNow(ImageView img, float deg) {
		img.setRotation(deg - 90);
	}



}

File: app/src/main/java/ds/features/binding/WeatherViewModel.kt
package ds.features.binding

import android.os.Bundle
import android.view.View
import ds.features.L
import ds.features.databinding.ActivityDetailsBinding
import ds.features.db.realm.Weather

class WeatherViewModel() : BaseViewModel<ActivityDetailsBinding>() {

	constructor(w: Weather) : this() {
		data = w
	}

	lateinit var data: Weather

	override fun onViewModelCreated() {
		super.onViewModelCreated()
		val id = view.bundle.getLong("weatherId", -1);
		data = db.getWeatherModel(id)
	}

	fun humidityVisibility(): Int {
		return if (data.humidity == null) View.GONE else View.VISIBLE
	}

	fun onClick(v: View) {
		L.v("click")
		val cx = (v.left + v.right) / 2
		val cy = (v.top + v.bottom) / 2
		val b = Bundle()
		b.putInt("cx",cx)
		b.putInt("cy",cy)
		b.putInt("w",v.width)
		view.navigate(1, b)

	}

	override fun toggleProgress(enable: Boolean) {
	}

}
